subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy

```diff
@@ -73,4 +77,27 @@ class BuildSrcApiChangePerformanceTest extends AbstractCrossVersionPerformanceTe
                 20
         }
     }
+
+    static Function<InvocationSettings, BuildMutator> createChangingClassInBuildSrcBeforeScenario(String filePath) {
```

Comment:
Can you define something like
```
static BuildMutator createChangingClass(InvocationSettings settings, String filePath) {
    return new BuildMutator { ... }
}
```

and then use this method via a method reference like
```
addBuildMutator { createChangingClass(it, changingClassFilePath) }
```

I think that would be better readable than this nested anonymous classes.

Another option would be to create a concrete mutator class with the invocaiton settings and the file path as constructor arguments and then add this.

Reply:
Done and done.